### PR TITLE
Issue 20 and 24

### LINKED ISSUE #20: src/minix/commands/autopart/autopart.c:1534: dead code block ?
[src/minix/commands/autopart/autopart.c:1534] -> [src/minix/commands/autopart/autopart.c:1535]: (warning) Opposite conditions in nested 'if' blocks lead to a dead code block.

    if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
        if (device >= 0) { close(device); device= -1; }
        return;
    }

### PR DESCRIPTION
**Pull request by Tookmund**
I have attempted to fix #20 and #24

### CODE REVIEW

**Comment by Tookmund** [comment]
Applied my changes using [the pull request patch] (https://github.com/Stichting-MINIX-Research-Foundation/minix/pull/27.patch)
Everything builds fine. Not sure how to test it though.

File: minix/commands/autopart/autopart.c [hunk 1]
```diff
@@ -1531,8 +1531,13 @@ void m_read(int ev, int *biosdrive)
...
 
-	if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
-		if (device >= 0) { close(device); device= -1; }
+	device = open(curdev->name, O_RDWR, 0666);
+	if (device < 0) {
+		return;
+	}
+	else if (device >= 0) {
+		close(device);
+		device = -1;
```
Comment (dcvmoole) [inline minix/commands/autopart/autopart.c]: This change is not correct, as it inverts the behavior of the code that was there in the first place. The original line 1535 can only trigger if device is set to a less-than-zero value, so all that needs to be done is removal of that particular line.

Comment (Tookmund) [inline minix/commands/autopart/autopart.c]: Oops! I've removed the else if block.
Do you want me to put it back to what is was before but without line 1535? It's only a couple extra lines and it is, IMO, much easier to read and understand.

Comment (dcvmoole) [inline minix/commands/autopart/autopart.c]: In this case I personally don't mind, but.. as a general rule, it's best not to fix multiple issues in a patch with a single purpose. Even if the unrelated changes improve the code, they make the patch harder to check and might have unintended side effects. For example, here you remove the assignment to the "mode" variable in line 1534, so I had to check whether that variable is used somewhere down the code. It is not, so your patch will now add a new "unused variable" compiler warning, and should actually also remove the declaration of that variable a bit higher up. All this has nothing to do with the original problem, which was just about line 1535, and as such these "ripple effects" complicate the reviewing process, well you get the point :)

Comment (Tookmund) [inline minix/commands/autopart/autopart.c]: Yeah sorry this pull request kinda exploded. I'll do my best to split up my pull requests better next time. At this point, I'm just going to remove mode as this is already a bit of a mess. Sorry about that. Thank you for taking the time to work on this!

**Comment by dcvmoole** [comment]
Sorry about the delay, apparently I'm not receiving all the email from github either (this spamfilter stuff is starting to be really annoying). The resulting patch looks good to me, but could you please squash the whole thing into one or two patches (one for autopart, one for libddekit)? There's no use merging the intermediate commits..

**Comment by sambuc** [comment]
I have submitted to the review system the fix for #24, you can follow up at:
  http://gerrit.minix3.org/#/c/2950/

I took the liberty of adding which issue it closes in the commit message, I hope it is fine with you.

I will do the same as soon as you have a merged commit for the proposals you have for #20.


Regards, 

Lionel

**Comment by dcvmoole** [comment]
Oops, I missed the fact that one of them was already a separate patch. Thanks @sambuc for looking better than I did and picking up the patch :)

**Comment by Tookmund** [comment]
Sorry I've been busy recently and haven't had a chance to look at this.  
I can try squashing the commits for #20 tomorrow. 
I don't mind that you changed the commit message feel free to do that any time my message missed something. 

Jacob

**Comment by Tookmund** [comment]
I've squashed the commits. The commit message is probably too long, feel free to shorten it.

**Comment by sambuc** [comment]
Hi, I have submitted your PR to gerrit:

  http://gerrit.minix3.org/2951

Thanks for your contribution!

**Comment by sambuc** [comment]
Both Issues are now closed. Thank you for your Contribution!

**Pull request closed without merging**